galaxy_importer/collection.py

```diff
@@ -173,12 +232,105 @@ def _load_collection_manifest(self):
+
+        files_manifest_file = os.path.join(path_prefix, file_manifest_file.name)
+        default_logger.debug('files_manifest_file: %s', files_manifest_file)
+
+        with open(files_manifest_file, 'r') as f:
+            try:
+                file_manifest = schema.CollectionArtifactFileManifest.parse(f.read())
+            except ValueError as e:
+                # TODO: Should this be a subclass of ValueError?
+                raise exc.ManifestValidationError(str(e))
+
+        return file_manifest
+
+    def _check_file_manifest(self, path_prefix, file_manifest, file_manifest_name):
+        """Check the file content described in file_manifest
+
+        Check the chksums for files.
+        Check for any missing files.
+        Check for any missing dirs.
+
+        Args:
+            path_prefix (str): Any file path prefix we need to add to file paths in the
+                CollectionArtifactFile artifact_file
+            file_manifest (CollectionArtifactFileManifest): Object with list of
+                CollectionArtifactFile items. The CollectionArtifactFile data
+                is used to check and validate each file.
+            file_manifest_name (str): The name of the file manifest (ie, "FILES.json")
+
+        Raises:
+            CollectionArtifactFileNotFound: If artifact_file is not found on the file system.
+            CollectionArtifactFileChecksumError: If the sha256sum of the on disk
+                artifact_file contents does not match artifact_file.chksum_sha256.
+
+        Returns:
+            bool: All the items in file_manifest were found and valid
+        """
+
+        for artifact_file in file_manifest.files:
+            if artifact_file.ftype != 'file':
+                continue
+
+            check_artifact_file(path_prefix=path_prefix,
+                                artifact_file=artifact_file)
+
+        # check the extract archive for any extra files.
+        ftw = FileWalker(collection_path=path_prefix)
+        prefix = path_prefix + '/'
+        found_file_set = set([string_utils.removeprefix(fp, prefix) for fp in ftw.walk()])
+
+        file_manifest_file_set = set([artifact_file.name for artifact_file in file_manifest.files])
+        # The artifact contains MANIFEST.json and FILES.JSON, but they aren't
+        # in file list in FILES.json so add them so we match expected.
+        file_manifest_file_set.add('MANIFEST.json')
+        file_manifest_file_set.add(file_manifest_name)
+
+        difference = sorted(list(found_file_set.difference(file_manifest_file_set)))
+        default_logger.debug('found_file_set.difference(file_manifest): %s',
+                             pf(difference))
+
+        if difference:
+            err_msg = \
+                f'Files in the artifact but not the file manifest: {difference}'
+            raise exc.FileNotInFileManifestError(unexpected_files=difference,
+                                                 msg=err_msg)
```

Comment:
Kind of want to extract this section to at least it's own method. It's kind of a hodgepodge of stuff at the moment.

Maybe even it's own module, so collection.py wouldn't need the FileWalker import or the string_utils import. 

hiding away some of the path munging and set() presentation also may make it clearer what `_check_file_manifest` does.